src/js/angular/clustermanagement/directives/cluster-list.directive.js

```diff
@@ -73,33 +73,45 @@ function ClusterListComponent($translate, $timeout, $repositories, productInfo,
 
             /**
              * Saves a node to the cluster based on the provided endpoint.
-             * If the node is available in the cluster, it is added, otherwise, a new location is created and added.
+             * If the node is found in the cluster deletion list, it is restored.
+             * If the node is available in the cluster, it is added.
+             * Otherwise, a new location is created and added.
+             *
              * @param {string} endpoint - The endpoint of the node to save.
              */
             $scope.saveNode = (endpoint) => {
-                const node = $scope.cluster.getAvailable().filter((available) => available.endpoint === endpoint)[0];
+                const findNodeByEndpoint = (nodes) => nodes.find((node) => node.endpoint === endpoint)[0];
```

Comment:
Why this logic is not in the cluster model?!

Reply:
done